tests/test.js

```diff
@@ -14,28 +15,67 @@ describe('crypter test', () => {
 })
 
 describe('WebUi Generator', () => {
-  let uniqueId = uuidV4()
   let email = 'fossasia@gmail.com'
-  let args = [
-    "-g", "https://github.com/fossasia/yaydoc.git",
-    "-t", "alabaster",
-    "-m", email,
-    "-u", uniqueId,
-    "-w", "true"
-  ]
-  let exitCode
-  let files
-
+  let data = {
+    email: email,
+    gitUrl: "https://github.com/sch00lb0y/yaydoc.git",
+    docTheme: ""
+  }
+  let generatorExitCode;
+  let githubPagesExitCode;
+  let herokuExitCode;
+  let files = [];
+  let uniqueId;
+  let noOfExit = 0;
+  let fakeSocket = (done) => {
+    return {
+      emit: (tag, data) => {
+        if (tag === 'github-success') {
+          githubPagesExitCode = 0;
+          noOfExit += 1;
+        } else if (tag === 'github-failure') {
+          githubPagesExitCode = data.errorCode;
+          noOfExit += 1;
+        } else if (tag === 'heroku-success') {
+          herokuExitCode = 0;
+          noOfExit += 1;
+        } else if (tag === 'heroku-failure') {
+          herokuExitCode = datat.errorCode;
+          noOfExit += 1;
+        }
+        if (noOfExit == 2) {
+          done();
+        }
+      }
+    }
+  }
   before((done) => {
-    let process = spawn('./generate.sh', args)
-    process.on('exit', (code) => {
-      exitCode = code
-      files = fs.readdirSync(`temp/${email}`)
-      done()
+    generator.executeScript({}, data, function(err, generatedData) {
+      if (err) {
+        generatorExitCode = err.exitCode;
+        done()
+      } else {
+        uniqueId = generatedData.uniqueId;
+        generatorExitCode = generatedData.exitCode;
+        files = fs.readdirSync(`temp/${email}`);
+        deploy.deployPages(fakeSocket(done), {
+          gitURL: "https://github.com/sch00lb0y/yaydoc.git",
+          encryptedToken: crypter.encrypt(process.env.GITHUB_OAUTH_TOKEN),
```

Comment:
yeah, that's why I'm testing in my local repository. In [open-event-android](https://github.com/fossasia/open-event-android/blob/development/upload-apk.sh#L15) also they been using GitHub token to push apk